### PR TITLE
DT-2013: 🐛 Handle possible duplicate noms numbers better

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/advice/SecureControllerAdvice.java [hunk 1]
```diff
@@ -62,7 +63,7 @@
 
     @ExceptionHandler(AccessDeniedException.class)
     public ResponseEntity<ErrorResponse> handleException(final AccessDeniedException e) {
-        log.debug("Forbidden (403) returned", e);
+        log.debug("Forbidden (403) returned with message: {}", e.getMessage());
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/controller/advice/SecureControllerAdvice.java:66]: stop exceptions appearing in `exceptions` in app insights unnecessarily

File: src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java [hunk 2]
```diff
@@ -98,37 +99,21 @@ public Custody updateCustodyPrisonLocation(final String nomsNumber,
                 updateCustody.getNomsPrisonInstitutionCode());
         return result.map(success -> {
             switch (success.outcome) {
-                case Updated:
-                    telemetryClient.trackEvent("P2PTransferPrisonUpdated", add(telemetryProperties, "updatedCount", String.valueOf(success.custodyRecordsUpdated.size())), null);
-                    break;
-                case NoUpdateRequired:
-                    telemetryClient.trackEvent("P2PTransferPrisonUpdateIgnored", telemetryProperties, null);
-                    break;
+                case Updated -> telemetryClient.trackEvent("P2PTransferPrisonUpdated", add(telemetryProperties, "updatedCount", String.valueOf(success.custodyRecordsUpdated.size())), null);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java:102]: intellij suggestion to switch to newer switch

File: src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java [hunk 3]
```diff
@@ -98,37 +99,21 @@ public Custody updateCustodyPrisonLocation(final String nomsNumber,
...
-                case MultipleCustodialSentences:
-                    telemetryClient.trackEvent("P2PTransferBookingNumberHasDuplicates", telemetryProperties, null);
-                    break;
-                case OffenderNotFound:
-                    telemetryClient.trackEvent("P2PTransferOffenderNotFound", telemetryProperties, null);
-                    break;
-                case MultipleOffendersFound:
-                    telemetryClient.trackEvent("P2PTransferMultipleOffendersFound", telemetryProperties, null);
-                    break;
+                case TransferPrisonNotFound -> telemetryClient.trackEvent("P2PTransferPrisonNotFound", telemetryProperties, null);
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java:111]: again intellij suggestion to auto change code

File: src/main/java/uk/gov/justice/digital/delius/controller/advice/SecureControllerAdvice.java [hunk 4]
```diff
@@ -97,7 +98,7 @@
 
     @ExceptionHandler(InvalidRequestException.class)
     public ResponseEntity<ErrorResponse> handleException(final InvalidRequestException e) {
-        log.debug("Bad Request (400) returned", e);
+        log.debug("Bad Request (400) returned with message: {}", e.getMessage());
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/digital/delius/controller/advice/SecureControllerAdvice.java:101]: Do we definitely want to suppress all 4XX exceptions?  I can see why for 401, 403, 404 as they are BAU, but 400/409 indicate the client has a problem which we might need to highlight?

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**